src/bin/pg_autoctl/ipaddr.c

```diff
@@ -274,20 +308,39 @@ fetchIPAddressFromInterfaceList(char *localIpAddress, int size)
 		return false;
 	}
 
-	for (ifa = ifaddr; ifa; ifa = ifa->ifa_next)
+	for (ifaddr = ifaddrList; ifaddr != NULL; ifaddr = ifaddr->ifa_next)
  	{
-		if (ifa->ifa_addr->sa_family == AF_INET && !(ifa->ifa_flags & IFF_LOOPBACK))
+		if (ifaddr->ifa_flags & IFF_LOOPBACK)
+		{
+			continue;
+		}
+
+		/*
+		 * We only support IPv4 here, also this function is only called in test
+		 * environment where we run in a docker container with a network
+		 * namespace in which we use only IPv4, so that's ok.
```

Comment:
maybe a good idea to have some tests run with ipv6, given the `SIGABORT` we ran into today.
Not something to do in this PR, but maybe open a ticket.